Author: Holger Macht <hmacht@suse.de>
Subject: ACPI: dock: Send key=value pair instead of plain value
Patch-mainline: yes
References: none

git commit 66b568218ab73be161dc109b913e6fa7dda44e16
and
git commit 79a8f70b4b9127eacfc91dd1436c4a7be05e62ab
merged together.

    Send key=value pair along with the uevent instead of a plain value so that
    userspace (udev) can handle it like common environment variables.
    
    Signed-off-by: Holger Macht <hmacht@suse.de>
    Acked-by: Kristen Carlson Accardi <kristen.c.accardi@intel.com>
    Cc: Stephan Berberig <s.berberig@arcor.de>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Acked-by: Greg Kroah-Hartman <gregkh@suse.de>
    Signed-off-by: Len Brown <len.brown@intel.com>

---
 drivers/acpi/dock.c |   18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

--- a/drivers/acpi/dock.c	2007-08-27 14:01:19.000000000 -0400
+++ b/drivers/acpi/dock.c	2007-08-27 14:01:21.000000000 -0400
@@ -328,11 +328,19 @@ static void hotplug_dock_devices(struct 
 static void dock_event(struct dock_station *ds, u32 event, int num)
 {
 	struct device *dev = &dock_device.dev;
-	/*
-	 * Indicate that the status of the dock station has
-	 * changed.
-	 */
-	kobject_uevent(&dev->kobj, KOBJ_CHANGE);
+	char event_string[13];
+	char *envp[] = { event_string, NULL };
+
+	if (num == UNDOCK_EVENT)
+		sprintf(event_string, "EVENT=undock");
+	else
+		sprintf(event_string, "EVENT=dock");
+
+        /*
+         * Indicate that the status of the dock station has
+         * changed.
+         */
+	kobject_uevent_env(&dev->kobj, KOBJ_CHANGE, envp);
 }
 
 /**
